client/auth.go

```diff
@@ -1,29 +1,43 @@
 package client
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 )
 
-func (c *Client) Auth() error {
+func (c *Client) Auth() (string, error) {
 	req, err := http.NewRequest("GET", fmt.Sprintf("https://%s/auth", c.Host), nil)
-
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	req.SetBasicAuth("convox", string(c.Password))
 
 	resp, err := c.client().Do(req)
-
 	if err != nil {
-		return err
+		return "", err
 	}
+	defer resp.Body.Close()
 
 	if resp.StatusCode != 200 {
-		return fmt.Errorf("invalid login\nHave you created an account at https://convox.com/signup?")
+		return "", fmt.Errorf("invalid login\nHave you created an account at https://convox.com/signup?")
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var id string
+	var data map[string]string
+
+	err = json.Unmarshal(body, &data)
+	if err == nil {
+		//if bad JSON is returned it is probably a legacy app
```

Comment:
questions 1 + 3) app is meant to be rack, i.e anything before this release, will update the comment

all racks will return "", nil - that is the current released behaviour (user ids are a console concept), i will add a comment